### PR TITLE
Only update synced_git_sha of track when syncing succeeds completely

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: test/commands/git/sync_track_test.rb [hunk 1]
```diff
@@ -27,6 +27,41 @@ class Git::SyncTrackTest < ActiveSupport::TestCase
     assert_equal git_track.head_sha, track.synced_to_git_sha
   end
 
+  test "git sync SHA does not change when concept syncing fails" do
+    track = create :track, slug: 'fsharp', active: true, synced_to_git_sha: "98403713252d41babae8353793ea5ec9ad7d770f"
+    Git::SyncConcept.expects(:call).raises(RuntimeError)
+
+    suppress(RuntimeError) do
```
Comment (ErikSchierboom) [inline test/commands/git/sync_track_test.rb]: I found this somewhere. Let me know if this is the right mechanism to ignore an error

Comment (iHiD) [reply test/commands/git/sync_track_test.rb]: Never come across `suppress` before. It's part of Rails FYI: https://api.rubyonrails.org/classes/Kernel.html#method-i-suppress

I normally use `assert_raises` to suppress, as that has the side effect of also ensuring the raising actually propagates to the top level. So I'd probably do that here too. Suppress is cool though - I'll use that in app code for sure.

Comment (iHiD) [inline test/commands/git/sync_track_test.rb]:
```suggestion
    assert_raises RuntimeError do
```

**Review by iHiD — APPROVED** [review]
This is good. Thanks!

File: test/commands/git/sync_track_test.rb [hunk 2]
```diff
@@ -27,6 +27,41 @@ class Git::SyncTrackTest < ActiveSupport::TestCase
...
+    end
+
+    assert_equal "98403713252d41babae8353793ea5ec9ad7d770f", track.synced_to_git_sha
+  end
+
+  test "git sync SHA does not change when concept exercise syncing fails" do
+    track = create :track, slug: 'fsharp', active: true, synced_to_git_sha: "98403713252d41babae8353793ea5ec9ad7d770f"
+    Git::SyncConceptExercise.expects(:call).raises(RuntimeError)
+
+    suppress(RuntimeError) do
```
Comment (ErikSchierboom) [inline test/commands/git/sync_track_test.rb]:
```suggestion
    assert_raises RuntimeError do
```

File: test/commands/git/sync_track_test.rb [hunk 3]
```diff
@@ -27,6 +27,41 @@ class Git::SyncTrackTest < ActiveSupport::TestCase
...
+    assert_equal "98403713252d41babae8353793ea5ec9ad7d770f", track.synced_to_git_sha
+  end
+
+  test "git sync SHA does not change when practice exercise syncing fails" do
+    skip # TODO: re-enable once we import practice exercises
+
+    track = create :track, slug: 'fsharp', active: true, synced_to_git_sha: "98403713252d41babae8353793ea5ec9ad7d770f"
+    Git::SyncPracticeExercise.expects(:call).raises(RuntimeError)
+
+    suppress(RuntimeError) do
```
Comment (ErikSchierboom) [inline test/commands/git/sync_track_test.rb]:
```suggestion
    assert_raises RuntimeError do
```

**Pull request merged**